python/python/glide/async_commands/core.py

```diff
@@ -2306,6 +2267,72 @@ async def geopos(
+        seach_by: Union[GeoSearchByRadius, GeoSearchByBox],
+        order_by: Optional[OrderBy] = None,
+        count: Optional[GeoSearchCount] = None,
+        with_coord: bool = False,
+        with_dist: bool = False,
+        with_hash: bool = False,
+    ) -> List[Union[str, List[Union[str, float, int, List[float]]]]]:
+        """
+        Searches for members in a sorted set stored at `key` representing geospatial data within a circular or rectengal area.
+
+        See https://valkey.io/commands/geosearch/ for more details.
+
+        Args:
+            key (str): The key of the sorted set representing geospatial data.
+            search_from (Union[str, GeospatialData]): The location to search from. Can be specified either as a member
+                from the sorted set or as a geospatial data (see `GeospatialData`).
+            search_by (Union[GeoSearchByRadius, GeoSearchByBox]): The search criteria.
+                For circular area search, see `GeoSearchByRadius`.
+                For rectengal area search, see `GeoSearchByBox`.
+            order_by (Optional[OrderBy]): Specifies the order in which the results should be returned.
+                Defaults to None. See `OrderBy`.
+            count (Optional[GeoSearchCount]): Specifies the maximum number of results to return.
+                Defaults to None. See `GeoSearchCount`.
+            with_coord (bool): Whether to include coordinates of the returned items. Defaults to False.
+            with_dist (bool): Whether to include distance from the center in the returned items.
+                The distance is returned in the same unit as specified for the `search_by` arguments. Defaults to False.
+            with_hash (bool): Whether to include geohash of the returned items. Defaults to False.
+
+        Returns:
+            List[Union[str, List[Union[str, float, int, List[float]]]]]: If no with_* was set to True, returns a list of memebers
+            names.
+            If `with_coord`, `with_dist` or `with_hash` was specified, returns an array of arrays, we're each sub array represents a single item with:
+                (str): The member name.
+                (float): The distance from the center as a floating point number, in the same unit specified in the radius, if `with_dist` is set to True.
+                (int) The Geohash integer, if `with_dist` is set to True.
+                List[float]: The coordinates as a two items x,y array (longitude,latitude), if `with_dist` is set to True.
+
+        Examples:
+            >>> await client.geoadd("my_geo_sorted_set", {"edge1": GeospatialData(12.758489, 38.788135), "edge2": GeospatialData(17.241510, 38.788135)}})
+            >>> await client.geoadd("my_geo_sorted_set", {"Palermo": GeospatialData(13.361389, 38.115556), "Catania": GeospatialData(15.087269, 37.502669)})
+            >>> await client.geosearch("my_location", "search_center", GeoSearchByRadius(5.0, GeoUnit.KILOMETERS))
```

Comment:
I think you need `with_coord=true, with_dist=true, with_hash=true`

Reply:
yes, thanks